packages/assets-controllers/src/TokensController.test.ts

```diff
@@ -198,6 +198,84 @@ describe('TokensController', () => {
     });
   });
 
+  it('should add tokens and update existing ones and detected tokens', async () => {
```

Comment:
nit - we can do this later in a future PR.

There is a lot of things happening in the controller/test. It would be nice to move/group some composable "arrange" functions to clean up these tests and make it clear what we are testing.